src/config.rs

```diff
@@ -57,10 +59,31 @@ impl FnmConfig {
     }
 
     pub fn base_dir_with_default(&self) -> std::path::PathBuf {
-        ensure_exists_silently(
-            (self.base_dir.clone())
-                .unwrap_or_else(|| home_dir().expect("Can't get home directory").join(".fnm")),
-        )
+        let user_pref = self.base_dir.clone();
+        if let Some(dir) = user_pref {
+            return dir;
+        }
+
+        let legacy = home_dir()
+            .map(|dir| dir.join(".fnm"))
+            .filter(|dir| dir.exists());
+        if let Some(dir) = legacy {
+            outln!(
+                self#Error,
+                "{} {} is deprecated. {} is now the default.",
+                "warning:".yellow().bold(),
+                "$HOME/.fnm".italic(),
+                "$XDG_DATA_HOME/fnm".italic()
+            );
```

Comment:
Let's use explicit directory paths in the warning so it will work on Windows as well. I think we should have a thorough explanation. Maybe even linking to the issue on GitHub?

```
warning:
  Looks like you have ${legacy} directory on your disk.
  fnm is migrating its default storage location from ${modern} to ${legacy}.
  Read more about it here: ${GH_ISSUE}

hint:
  You can use the following command to move the directory:
    mv ${legacy} ${modern}
  Please make sure your `fnm` binary is not stored in this directory and if it does, move it to ${A LOCATION WE NEED TO CHOOSE AND UPDATE THE INSTALLATION SCRIPT TO USE AS WELL} and update your shell profile accordingly.
```